src/utils/trip.js

```diff
@@ -47,3 +49,50 @@ export const sortTime = (pointA, pointB) => {
 export const sortPrice = (pointA, pointB) => {
   return pointB.price - pointA.price;
 };
+
+const countTimeSpendByType = (points, type) => {
+  const pointsByType = points.filter((point) => point.type === type);
+  if(pointsByType.length!=0){
+    return pointsByType.reduce((prev, curr) => {return prev + dayjs.duration(dayjs(curr.dateTimeEnd).diff(dayjs(curr.dateTimeStart))).asMinutes();}, 0);
+  }else {
+    return dayjs.duration(0, 'minutes').asMinutes();
+  }
+};
+
+const countTransportByType = (points, type) => {
+  const transportByType = points.filter((point) => point.type === type);
+  return transportByType.length;
+};
+
+export const countTimeSpendByTypes = (points) => {
+  const times =[];
+  TYPES.forEach((element) => times.push(
+    {
+      time: Math.abs(countTimeSpendByType(points,element.name)),
+      name: element.name.toUpperCase(),
+    }));
+  const res = times.sort((elementA, elementB) => {return elementB.time - elementA.time;});
+  return res;
+};
+
+export const countPointsMoneyByTypes = (points) => {
+  const prices =[];
+  TYPES.forEach((element) => prices.push(
+    {
+      price: Math.round(Math.abs(countTimeSpendByType(points,element.name))),
+      name: element.name.toUpperCase(),
+    }));
+  const res = prices.sort(sortPrice);
+  return res;
+};
+
+export const countTransportByTypes = (points) => {
+  const transport =[];
+  TYPES.forEach((element) => transport.push(
+    {
+      count: countTransportByType(points,element.name),
+      name: element.name.toUpperCase(),
+    }));
+  const res = transport.sort((elementA, elementB) => {return elementB.count - elementA.count;});
```

Comment:
Можно использовать короткий синтаксис для стрелочных функций возвращающих значение.
```suggestion
  const res = transport.sort((elementA, elementB) => elementB.count - elementA.count);
```